src/java.base/share/classes/java/lang/foreign/FunctionDescriptor.java

```diff
@@ -94,6 +95,22 @@ public sealed interface FunctionDescriptor permits FunctionDescriptorImpl {
      */
     FunctionDescriptor dropReturnLayout();
 
+    /**
+     * Returns the method type consisting of the carrier types of the layouts in this function descriptor.
+     * <p>
+     * The carrier type of a layout is determined as follows:
+     * <ul>
+     * <li>If the layout is a {@link ValueLayout} the carrier type is determined through {@link ValueLayout#carrier()}.</li>
+     * <li>If the layout is a {@link GroupLayout}, or {@link SequenceLayout} the carrier type is {@link MemorySegment}.</li>
+     * <li>If the layout is a {@link PaddingLayout} an {@link IllegalArgumentException} is thrown.</li>
+     * </ul>
+     *
+     * @return the method type consisting of the carrier types of the layouts in this function descriptor
+     * @throws IllegalArgumentException if one or more layouts in the function descriptor are not supported
+     * (e.g. if they are sequence layouts or padding layouts).
+     */
+    MethodType carrierMethodType();
```

Comment:
this name seems long, and the extra wording doesn't seem to add a lot (at least IMHO).